### PR TITLE
[codex] PR 07 — Risk / Leverage module

### PR DESCRIPTION
**Pull request by pivox**
## Objectif

Introduire/formaliser le module Risk / Leverage du TradingCore.

Cette PR clarifie la source de vérité du risque, les champs legacy, le calcul cible du levier dérivé et les caps, sans changer le comportement runtime ni les valeurs de risque/levier.

## Scope

- Ajout de DTOs / services Risk / Leverage.
- Ajout de tests unitaires sur risk/leverage.
- Documentation du module Risk / Leverage.
- Aucun changement runtime sauf délégation mécanique sûre si nécessaire.

## Hors-scope

- Aucun changement `mtf:run`.
- Aucun changement `POST /api/mtf/run`.
- Aucun changement Temporal.
- Aucun changement règles MTF.
- Aucun changement READY / REJECTED.
- Aucun changement EntryZone.
- Aucun changement SL/TP.
- Aucun branchement `EffectiveTradingConfigResolver`.
- Aucun live OKX/Hyperliquid.
- Aucune suppression Bitmart.
- Aucun changement YAML stratégie.
- Aucun changement des valeurs risk/leverage.
- Aucun objectif d’augmentation du risque ou du levier.

## Tests

- `vendor/bin/phpunit --bootstrap vendor/autoload.php tests/TradingCore/Risk`
- `vendor/bin/phpunit --bootstrap vendor/autoload.php tests/TradingCore/Risk tests/TradingCore/Entry tests/TradingCore/Mtf tests/TradingCore/Decision tests/Application/Runner`
- `vendor/bin/phpstan analyse src/TradingCore/Risk tests/TradingCore/Risk --memory-limit=512M`
- `vendor/bin/phpstan analyse src/TradingCore/Risk src/TradingCore/Entry src/TradingCore/Mtf src/TradingCore/Decision tests/TradingCore/Risk tests/TradingCore/Entry tests/TradingCore/Mtf tests/TradingCore/Decision --memory-limit=512M`
- `DEFAULT_URI=http://localhost:8082 php bin/console lint:yaml config`
- `DEFAULT_URI=http://localhost:8082 php bin/console lint:container`
- `DEFAULT_URI=http://localhost:8082 php bin/console list mtf --raw | grep -E '^mtf:run\s'`
- `DEFAULT_URI=http://localhost:8082 php bin/console debug:router --show-controllers | grep -E '/api/mtf/run|mtf_run'`
- `python3 -m mkdocs build --strict`
- `git diff --check`

## Risques

Le risque principal est de modifier involontairement la taille de position ou le levier effectif. Cette PR reste une formalisation/préparation sans changement de comportement runtime.

Note locale: les commandes Symfony nécessitent `DEFAULT_URI=http://localhost:8082` dans cet environnement. `lint:container` passe mais affiche une dépréciation existante sur `EntryZoneStatsCommand::__construct()`.

## Critères d’acceptation

- Module Risk / Leverage ajouté.
- Tests risk/leverage ajoutés.
- Source de vérité du risque documentée.
- Aucun changement de valeur risk/leverage.
- Aucun levier arbitraire.
- Entrypoints préservés.
- Aucun comportement runtime changé.

### CODE REVIEW

File: trading-app/src/TradingCore/Risk/Service/RiskConfigInterpreter.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                'warnings' => $warnings,
+            ],
+        );
+    }
+
+    public function normalizePercent(float $value): float
+    {
+        $value = max(0.0, $value);
+        if ($value > 1.0) {
+            $value *= 0.01;
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/RiskConfigInterpreter.php:90]: l'assertion de Codex ("existing fixed-risk accessors divide by 100 unconditionally") est incorrecte — RiskConfigInterpreter est le seul
  lecteur de fixed_risk_pct et vient d'être créé dans cette PR. Tous les YAML ont des valeurs > 1.0 (5.0, 2.0, 1.5). La convention > 1.0 = pourcentage est intentionnelle et cohérente avec
  PR06 EntryZone. J'ai ajouté un docblock sur normalizePercent pour que ce choix soit explicite et ne surprenne pas les futurs contributeurs.

File: trading-app/src/TradingCore/Risk/Service/PositionSizer.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+        throw new \InvalidArgumentException('risk pct source is required');
+    }
+
+    private function resolveCapitalBase(RiskCalculationRequest $request, float $riskPct): float
+    {
+        $initialMargin = $request->initialMarginUsdt;
+        if ($initialMargin !== null && $initialMargin > 0.0) {
+            if ($request->availableBalance !== null && $request->availableBalance > 0.0) {
+                return min($initialMargin, $request->availableBalance);
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/PositionSizer.php]: Corrigé en 6a57786. La condition `> 0.0` est remplacée par `!== null` : `min(initialMargin, max(0.0, availableBalance))`. Un `availableBalance = 0.0` explicite donne `capitalBase = 0` → exception `capital base must be positive`, ce qui correspond au comportement legacy `OrderPlanBuilder` (`min(initial_margin_usdt, max(available_usdt, 0))`). Test ajouté : `testRejectsPositionWhenAvailableBalanceIsExplicitlyZero`.

File: trading-app/src/TradingCore/Risk/Service/LeverageCalculator.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            // Floor cannot bypass exchange/profile/symbol caps: cap it to cappedLeverage first.
+            $bounded = max($bounded, min($request->floor, $cappedLeverage));
+        }
+
+        $rounded = $this->round($bounded, $request->roundingMode);
+        $rounded = max(1, $rounded);
+        $rounded = max($request->minLeverage, $rounded);
+        $rounded = min($request->maxLeverage, $rounded);
+        // Prevent ceil/round from pushing above a fractional float cap (e.g. cap=5.5, ceil→6).
+        $rounded = min($rounded, (int)floor($cappedLeverage));
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/LeverageCalculator.php]: Corrigé en 6a57786. Le clamp `min(rounded, floor(cappedLeverage))` était inconditionnel et s'appliquait aussi quand aucun cap n'avait réduit le levier (ex. `rawLeverage=1.5`, `exchangeCap=100` → `ceil(1.5)=2` clippé à 1). Le guard est maintenant conditionnel : il ne s'active que si `cappedLeverage < preCapLeverage` (un cap a effectivement réduit la valeur). Test ajouté : `testCeilRoundingIsPreservedWhenNoCapsReducedLeverage`.

File: trading-app/src/TradingCore/Risk/Service/PositionSizer.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            }
+
+            return $initialMargin;
+        }
+
+        if ($request->fallbackAccountBalance !== null && $request->fallbackAccountBalance > 0.0) {
+            // Mirrors TradeEntryRequestBuilder: initialMargin = fallbackCapital * riskPct.
+            // Caller applies effectiveRiskPct again → riskUsdt = balance * riskPct².
+            // All current YAML configs set fallback_account_balance: 0.0 so this path is unreachable in production.
+            return $request->fallbackAccountBalance * $riskPct;
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/PositionSizer.php]: Corrigé en 86a6a59. Le path `fallbackAccountBalance` applique maintenant `min(syntheticMargin, max(0.0, availableBalance))` quand `availableBalance !== null`, identique à ce qui a été fait pour le path `initialMarginUsdt`. Test ajouté : `testFallbackCapitalPathCappsToAvailableBalance`.

File: trading-app/src/TradingCore/Risk/Service/LeverageCalculator.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+        $rounded = $this->round($bounded, $request->roundingMode);
+        $rounded = max(1, $rounded);
+        $rounded = max($request->minLeverage, $rounded);
+        $rounded = min($request->maxLeverage, $rounded);
+        // Prevent ceil/round from pushing above a fractional float cap, but only when a cap
+        // actually reduced the leverage (cappedLeverage < preCapLeverage).
+        // When no cap was binding, cappedLeverage equals the raw leverage and the rounding
+        // mode must be respected as-is (e.g. rawLeverage=1.5, ceil→2 must stay 2).
+        if ($cappedLeverage < $preCapLeverage && $rounded > (int)floor($cappedLeverage)) {
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/LeverageCalculator.php]: Corrigé en 86a6a59. Le guard `cappedLeverage < preCapLeverage` est remplacé par une boucle sur chaque cap configurée : `foreach ([exchangeCap, profileCap, symbolCap] as $cap) { if ($rounded > $cap) { $rounded = floor($cap); } }`. Ça couvre l'edge case `rawLeverage == cap == 5.5` (cappedLeverage = preCapLeverage, ancienne condition ne firait pas). Les calculs sans cap binding restent inchangés car `rounded > 100` (ou similaire) n'est jamais vrai. Test ajouté : `testCeilRoundingCannotExceedCapWhenRawLeverageEqualsCapExactly`.

File: trading-app/src/TradingCore/Risk/Service/PositionSizer.php [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            }
+
+            return $syntheticMargin;
+        }
+
+        if ($request->availableBalance !== null && $request->availableBalance > 0.0) {
+            return $request->availableBalance;
+        }
+
+        return $request->equity !== null && $request->equity > 0.0 ? $request->equity : 0.0;
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/PositionSizer.php:127]: Corrigé en dc3bcf1. La condition `> 0.0` sur la branche `availableBalance/equity` est remplacée par `!== null` + `max(0.0, ...)` — cohérent avec les deux branches précédentes (`initialMarginUsdt`, `fallbackAccountBalance`). `availableBalance = 0.0` avec `equity = 500` retourne maintenant `capitalBase = 0` → exception `capital base must be positive`. Test ajouté : `testRejectsWhenAvailableBalanceIsZeroAndEquityIsPositive`.

File: trading-app/src/TradingCore/Risk/Service/PositionSizer.php [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+     */
+    private function resolveRiskPct(RiskCalculationRequest $request): array
+    {
+        $warnings = [];
+        $legacyRuntimeSource = $request->metadata['legacy_runtime_risk_source'] ?? null;
+        if (
+            $legacyRuntimeSource === RiskSource::RiskPctPercentLegacy
+            && $request->riskPctPercentLegacy !== null
+            && $request->riskPctPercentLegacy > 0.0
+        ) {
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/TradingCore/Risk/Service/PositionSizer.php:71]: Fixed in 6deb446.

Removed the `> 0.0` guard from the outer `RiskPctPercentLegacy` branch in `resolveRiskPct`. When `riskPctPercentLegacy` is set (even to 0.0), it now always returns that value with the legacy source. The existing `effectiveRiskPct <= 0.0` guard in `calculate()` then throws `InvalidArgumentException('effective risk pct must be positive')`, which mirrors the legacy `TradeEntryRequestBuilder` that returns `null` (i.e., rejects the trade) when `riskPct <= 0`.

Added regression test `testRejectsWhenLegacyRiskPctIsZeroInsteadOfFallingBackToFixedRisk`: `risk_pct_percent=0` + `fixed_risk_pct=5.0` → throws, not a sized trade.
